clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -1062,46 +1092,31 @@ invariant_checker_t::print_results()
 
 std::string
 invariant_checker_t::check_for_pc_discontinuity(
-    per_shard_t *shard, const memref_t &memref, const memref_t &prev_instr, addr_t cur_pc,
-    const std::unique_ptr<instr_autoclean_t> &cur_instr_decoded, bool expect_encoding,
+    per_shard_t *shard, const per_shard_t::instr_info_t &prev_instr_info,
+    const per_shard_t::instr_info_t &cur_instr_info, bool expect_encoding,
```

Comment:
But at a kernel event there is no current instruction: we're at a marker.  It seems a little weird to pass in "cur_instr_info" for a non-instruction.